src/Compilers/Core/Portable/InternalUtilities/ReadOnlyMemoryOfCharComparer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class ReadOnlyMemoryOfCharComparer : IEqualityComparer<ReadOnlyMemory<char>>
+    {
+        public static ReadOnlyMemoryOfCharComparer Instance { get; } = new();
+
+        private ReadOnlyMemoryOfCharComparer()
+        {
+        }
+
+        public bool Equals(ReadOnlyMemory<char> x, ReadOnlyMemory<char> y)
+            => x.Span.SequenceEqual(y.Span);
+
+        public int GetHashCode(ReadOnlyMemory<char> obj)
+        {
+#if NET
+            return string.GetHashCode(obj.Span);
+#else
+            return Hash.GetFNVHashCode(obj.Span);
+#endif
```

Comment:
unfortunately, netstandard has no good hash function for ROM<char>.  So we just defer to a reasonable one that roslyn has.